cmake/BuildLibSanitizers.cmake

```diff
@@ -23,36 +22,58 @@ macro(GEN_BUILD_TARGET name libsanitizer_path libfuzzer_path
 
   # Strip preinit object files in static libraries, otherwise a message
   # `.preinit_array section is not allowed in DSO` will prevent building DSO.
+  if(strip AND NOT "${strip}" STREQUAL "")
+    set(STRIP_COMMAND ${CMAKE_C_COMPILER_AR} d ${libsanitizer_name} ${strip})
+    set(STRIP_COMMENT "Strip sanitizer library ${name}")
+    set(STRIP_BYPRODUCTS BYPRODUCTS ${libsanitizer_name})
+  else()
+    set(STRIP_COMMAND ${CMAKE_COMMAND} -E echo "Skip stripping sanitizer library ${name} - no objects to strip")
+    set(STRIP_COMMENT "Skip stripping sanitizer library ${name} (no objects to strip)")
+    set(STRIP_BYPRODUCTS "")
+  endif()
```

Comment:
Probably it is better to set `DEPENDS` to empty string for `build_dso_*` targets than for every `strip_lib_*` target.
BTW, why stripping is not required on macOS?

Reply:
On macOS debug info is stored in separate `.dSYM` bundles and since they are already extracted during the build, no need to strip them separately

Reply:
`.preinit_array` is not about debug info, it is a list of function pointers that are executed before the `main _init` function. `.preinit_array` is not allowed in DSO, so we remove them, see explanation in the appropriate commit - https://github.com/ligurio/luzer/commit/bd91d593bb0efc54efd7c07c0287b991d2a911e3